src/main/java/teammates/ui/template/InstructorCommentsCommentRow.java

```diff
@@ -0,0 +1,61 @@
+package teammates.ui.template;
+
+import teammates.common.datatransfer.CommentAttributes;
+
+public class InstructorCommentsCommentRow extends CommentRow{
+    private Boolean isInstructorAllowedToModifyCommentInSection;
+    private String typeOfPeopleCanViewComment;
+    private String editedAt;
+    private VisibilityCheckboxes visibilityCheckboxes;
+    
+    // Visibility Settings
+    private String showCommentsTo;
+    private String showGiverNameTo;
+    private String showRecipientNameTo;

```

Comment:
`visibilityCheckboxes` and the `show*To` variables are related, both are derived from `comment.show*To` and are two different representations of the visibility settings of the comment.

`visibilityCheckboxes` shows whether each comment participant has permissions to view comments/giverName/recipientName while `show*To` shows the participants allowed to view comments/giverName/recipientName. 

Similarly for `typeOfPeopleCanViewComment` and `showCommentsTo`, `typeOfPeopleCanViewComment` is a description of the people who can view the comment (as shown below)

![view](https://cloud.githubusercontent.com/assets/9999077/8201856/5aea43c4-1507-11e5-9f35-5433b85198a8.PNG)

while `showCommentsTo` is a string representation of the `Enum` of the `CommentParticipantType`s who can view the comments.

By combining them, do you mean that rather than passing them as params into the constructor, I should derive them in the constructor instead? Wouldn't that mean introducing more logic into the class though. If I pass them in as params I could just use the methods available in the pageData class.